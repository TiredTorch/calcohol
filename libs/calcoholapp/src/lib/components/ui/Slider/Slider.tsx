import { FC } from "react";
import { Slider as MUISlider, SliderProps } from "@mui/material";

export const Slider: FC<SliderProps> = ({
	...rest
}) => {
	const marks = [
		{
			value: 0,
			label: "4:0",
		},
		{
			value: 25,
			label: "3:1",
		},
		{
			value: 50,
			label: "2:2",
		},
		{
			value: 75,
			label: "1:3",
		},
		{
			value: 100,
			label: "0:4",
		},
	];
	return (
		<MUISlider
			step={null}
			marks={marks}
			{...rest}
		/>
	);
};

export default Slider;