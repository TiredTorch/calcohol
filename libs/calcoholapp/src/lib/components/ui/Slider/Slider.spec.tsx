import "@testing-library/jest-dom";
import { Slider } from "./Slider";
import { render } from "@testing-library/react";

afterEach(() => {
	jest.resetAllMocks();
});

it("should render slider", () => {
	const { baseElement } = render(
		<Slider />
	);

	expect(baseElement).toBeInTheDocument();
	expect(baseElement).toMatchSnapshot();
});
