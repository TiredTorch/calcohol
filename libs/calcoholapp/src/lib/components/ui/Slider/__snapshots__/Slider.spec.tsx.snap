// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render slider 1`] = `
<body>
  <div>
    <span
      class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-marked css-1aznpnh-MuiSlider-root"
    >
      <span
        class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
      />
      <span
        class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
        style="left: 0%; width: 0%;"
      />
      <span
        class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index="0"
        style="left: 0%;"
      />
      <span
        aria-hidden="true"
        class="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
        data-index="0"
        style="left: 0%;"
      >
        4:0
      </span>
      <span
        class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index="1"
        style="left: 25%;"
      />
      <span
        aria-hidden="true"
        class="MuiSlider-markLabel css-yafthl-MuiSlider-markLabel"
        data-index="1"
        style="left: 25%;"
      >
        3:1
      </span>
      <span
        class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index="2"
        style="left: 50%;"
      />
      <span
        aria-hidden="true"
        class="MuiSlider-markLabel css-yafthl-MuiSlider-markLabel"
        data-index="2"
        style="left: 50%;"
      >
        2:2
      </span>
      <span
        class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index="3"
        style="left: 75%;"
      />
      <span
        aria-hidden="true"
        class="MuiSlider-markLabel css-yafthl-MuiSlider-markLabel"
        data-index="3"
        style="left: 75%;"
      >
        1:3
      </span>
      <span
        class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index="4"
        style="left: 100%;"
      />
      <span
        aria-hidden="true"
        class="MuiSlider-markLabel css-yafthl-MuiSlider-markLabel"
        data-index="4"
        style="left: 100%;"
      >
        0:4
      </span>
      <span
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium css-eg0mwd-MuiSlider-thumb"
        data-focusvisible="false"
        data-index="0"
        style="left: 0%;"
      >
        <input
          aria-orientation="horizontal"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          data-index="0"
          max="100"
          min="0"
          style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
          type="range"
          value="0"
        />
      </span>
    </span>
  </div>
</body>
`;
